pkg/client/rest_client_test.go

```diff
@@ -0,0 +1,55 @@
+package client_test
+
+import (
+	"net/http"
+	"testing"
+
+	restclient "github.com/codeready-toolchain/toolchain-common/pkg/client"
+	clienttest "github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/h2non/gock.v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestCreateTokenRequest(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		// given
+		const apiEndpoint = "https://api.example.com"
+		clienttest.SetupGockForServiceAccounts(t, apiEndpoint, types.NamespacedName{
+			Name:      "jane",
+			Namespace: "jane-env",
+		})
+		cl, err := clienttest.NewRESTClient("secret_token", apiEndpoint)
+		cl.Client.Transport = gock.DefaultTransport // make sure that the underlying client's request are intercepted by Gock
+
+		// when
+		require.NoError(t, err)
+		token, err := restclient.CreateTokenRequest(cl, types.NamespacedName{
+			Namespace: "jane-env",
+			Name:      "jane",
+		}, 1)
+
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, "token-secret-for-jane", token) // `token-secret-for-jane` is the answered mock by Gock in `clienttest.SetupGockForServiceAccounts(...)`
+	})
+	t.Run("failure", func(t *testing.T) {
+		// given
+		// the api server returns an error an a nil token request
+		const apiEndpoint = "https://api.example.com"
+		clienttest.SetupGockWithCleanup(t, apiEndpoint, "api/v1/path/", "", http.StatusInternalServerError)
+		cl, err := clienttest.NewRESTClient("secret_token", apiEndpoint)
+
+		// when
+		require.NoError(t, err)
+		token, err := restclient.CreateTokenRequest(cl, types.NamespacedName{
+			Namespace: "jane-env",
+			Name:      "jane",
+		}, 1)
+
+		// then
+		require.Error(t, err)      // an error should be returned
```

Comment:
Let's also check the error message, to make sure it's the actual error we expect here and not something else.

Reply:
Actually seems that this can never happen , see also linter issue:
```
pkg/client/rest_client.go:29:5: SA4031: this nil check is never true (staticcheck)
	if result == nil {
	   ^
```
removed that check https://github.com/codeready-toolchain/toolchain-common/pull/310/commits/55fc7e763c742def67bb7bc1f0ae10de667db077